Added core/utility.hpp include

steevo87 opened the pull request:
Necessary to enable calls such as "cv::getTickCount()"

steevo87 committed:
Added core/utility.hpp include

Necessary to enable calls such as "cv::getTickCount()"

kwaegel commented:
Is there any reason to add this header without any corresponding code that uses it?

steevo87 commented:
I had to include the header for the original code to compile. Without it, I was getting these errors:

2>C:\Users\ur10\Documents\GitHub\kinfu_remake\kfusion\src\core.cpp(206): error C2039: 'getTickCount' : is not a member of 'cv'
2>C:\Users\ur10\Documents\GitHub\kinfu_remake\kfusion\src\core.cpp(206): error C3861: 'getTickCount': identifier not found
2>C:\Users\ur10\Documents\GitHub\kinfu_remake\kfusion\src\core.cpp(222): error C2039: 'getTickCount' : is not a member of 'cv'
2>C:\Users\ur10\Documents\GitHub\kinfu_remake\kfusion\src\core.cpp(222): error C2039: 'getTickFrequency' : is not a member of 'cv'
2>C:\Users\ur10\Documents\GitHub\kinfu_remake\kfusion\src\core.cpp(222): error C3861: 'getTickCount': identifier not found
2>C:\Users\ur10\Documents\GitHub\kinfu_remake\kfusion\src\core.cpp(222): error C3861: 'getTickFrequency': identifier not found
2>C:\Users\ur10\Documents\GitHub\kinfu_remake\kfusion\src\core.cpp(227): error C2039: 'getTickCount' : is not a member of 'cv'
2>C:\Users\ur10\Documents\GitHub\kinfu_remake\kfusion\src\core.cpp(227): error C3861: 'getTickCount': identifier not found
2>C:\Users\ur10\Documents\GitHub\kinfu_remake\kfusion\src\core.cpp(231): error C2039: 'getTickCount' : is not a member of 'cv'
2>C:\Users\ur10\Documents\GitHub\kinfu_remake\kfusion\src\core.cpp(231): error C2039: 'getTickFrequency' : is not a member of 'cv'
2>C:\Users\ur10\Documents\GitHub\kinfu_remake\kfusion\src\core.cpp(231): error C3861: 'getTickCount': identifier not found
2>C:\Users\ur10\Documents\GitHub\kinfu_remake\kfusion\src\core.cpp(231): error C3861: 'getTickFrequency': identifier not found

kwaegel commented:
I just double checked, and `<opencv2/core/core.hpp>` is already included in `kfusion/types.hpp`. The include chain (`core.cpp`<`kinfu.hpp`<`types.hpp`) should already have included that for you, so I suspect something else is wrong with your build environment.

steevo87 commented:
Oh OK, so I'm guessing your `<opencv2/core/core.hpp>` includes a line like this:

```
#include <opencv2/core/utility.hpp>
```

Mine doesn't. I'm using the trunk version of OpenCV (last edited: 2014-04-11 19:18:04). Is there a chance that they removed that line of code in between the version of OpenCV you are using and the one I am? In my version of OpenCV I can only find a reference to `utility.hpp` in `core_c.h`

kwaegel commented:
That's probably it. I think trunk is in the middle of a huge refactoring for OpenCV 3.0. Try using the  OpenCV 2.4.9 tag that was recently added.

I last built this with OpenCV commit Itseez/opencv@a8cb5c39837cd4c968ff1f02aaca0e0ddcb3c424 from the 2.4 branch.

steevo87 committed:
Configuration improvements

Should adapt better to different types of OpenCV installations (e.g.
from pre-compiled binaries or local builds). Should default to correct
OpenNI 2 address. Changes untested on UR10-PC.

steevo87 committed:
CMake improvements

Better at finding and handling different versions of OpenCV, and now
copies the necessary DLLs/drivers to run the demo.

steevo87 committed:
Minor correction

REQUIRED listed twice in CMake line..

steevo87 committed:
Minor CMake change

Improved OpenCV location hints

steevo87 committed:
Added compatibility with trunk opencv

Includes an extra file if OpenCV version is greater than 2.

steevo87 committed:
Improved OpenCV cmake-ing

To handle trunk version better.

steevo87 committed:
Merge remote-tracking branch 'origin/master'

Conflicts:
	CMakeLists.txt
